catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscription.java

```diff
@@ -0,0 +1,184 @@
+        this.registration = null;
+    }
+
+    public SubscriptionMetadata getMetadata() {
+        return metadata;
+    }
+
+    public Optional<Subscription> getSubscription() {
+        return Optional.ofNullable(subscription);
+    }
+
+    public boolean isType(String type) {
+        return type.equals(metadata.getTypeName());
+    }
+
+    public boolean isNotType(String type) {
+        return !isType(type);
+    }
+
+    public boolean isNotRegistered() {
+        return registration == null;
+    }
+
+    /**
+     * Attempt to construct a {@link Subscription} using the provided factory. If the resultant
+     * subscription is valid, register it as this cached subscription's OSGi service.
+     *
+     * @param factory subscription factory corresponding to this cached subscription's type.
+     * @throws SubscriptionRegistrationException if the factory can't operate on this cached subscription.
+     */
+    public void registerSubscription(SubscriptionFactory factory) {
+        if (isNotType(factory.getTypeName())) {
+            throw new SubscriptionRegistrationException(format(
+                    "Factory type mismatch for subscription type [%s]",
+                    metadata.getTypeName()));
+        }
+
+        LOGGER.debug("Regenerating subscription [ {} | {} | {} ]",
+                metadata.getId(),
+                metadata.getTypeName(),
+                metadata.getCallbackAddress());
+
+        Subscription subscription = factory.createSubscription(metadata);
+        if (subscription == null) {
+            throw new SubscriptionRegistrationException("Regenerated subscription cannot be null");
+        }
+        registerSubscription(subscription);
+    }
+
+    /**
+     * Register the provided {@link Subscription} as this cached subscription's OSGi service.
+     *
+     * @param subscription the subscription to register.
+     * @throws SubscriptionRegistrationException if registration failed.
+     */
+    public void registerSubscription(Subscription subscription) {
+        LOGGER.debug("Registering service [{}]", metadata.getId());
+
+        Dictionary<String, String> properties = new Hashtable<>();
+        properties.put(SUBSCRIPTION_ID_OSGI, metadata.getId());
+        properties.put(EVENT_ENDPOINT, metadata.getCallbackAddress());
+
+        registration = getBundleContext().registerService(Subscription.class.getName(),
+                subscription,
+                properties);
+
+        if (registration != null) {
+            LOGGER.debug("Subscription [ {} | {} | {} ] registered with bundle ID = {} ",
+                    metadata.getId(),
+                    metadata.getTypeName(),
+                    metadata.getCallbackAddress(),
+                    registration.getReference()
+                            .getBundle()
+                            .getBundleId());
+            this.subscription = subscription;
+        } else {
+            throw new SubscriptionRegistrationException(format(
```

Comment:
This code has changed from last night, subscription was being set before registration and left non-null. I meant resetting subscription to null and not registration.